src/scripts/store/dimStoreItem.directive.js

```diff
@@ -91,6 +91,8 @@ function StoreItem(dimItemService, dimStoreService, ngDialog, dimLoadoutService,
     vm.clicked = function openPopup(item, e) {
       e.stopPropagation();
 
+      $rootScope.$broadcast('item-clicked', item);
```

Comment:
Upon further reflection it isn't too bad, but note that Promises are always async so you can't go wrong there.